### PR TITLE
feat: Adding basic error type support for output units.

### PR DESCRIPTION
**Pull request by CheesyBoy123**
Motivation:

As described in https://github.com/eclipse-vertx/vertx-json-schema/issues/132 we want to add support for OutputErrorTypes. Whenever we have an error there should be *some* kind of error type associated. For instance if the field is required but is missing we should return back the error type of MISSING_VALUE. 

For recursive error checks like $ref, $anyOf, $not etc, the error type is taken from the recursive OutputUnit.
For primitive checks like array length checks, format validation, required checks etc I have added what I believe the correct Error Type should be. 

Conformance:

You should have signed the Eclipse Contributor Agreement as explained in https://github.com/eclipse/vert.x/blob/master/CONTRIBUTING.md
Please also make sure you adhere to the code style guidelines: https://github.com/vert-x3/wiki/wiki/Vert.x-code-style-guidelines

### CODE REVIEW

File: src/main/java/io/vertx/json/schema/impl/SchemaValidatorImpl.java [hunk 1]
```diff
@@ -599,7 +600,7 @@ private OutputUnit validate(final Object _instance, final JsonSchema schema, fin
                 thisEvaluated.add(key);
               } else {
                 stop = outputFormat == OutputFormat.Flag;
-                errors.add(new OutputUnit(subInstancePointer, computeAbsoluteKeywordLocation(schema, schemaLocation + "/patternProperties"), baseLocation + "/patternProperties", "Property \"" + key + "\" matches pattern \"" + pattern + "\" but does not match associated schema"));
+                errors.add(new OutputUnit(subInstancePointer, computeAbsoluteKeywordLocation(schema, schemaLocation + "/patternProperties"), baseLocation + "/patternProperties", "Property \"" + key + "\" matches pattern \"" + pattern + "\" but does not match associated schema", OutputErrorType.MISSING_VALUE));
```
Comment (pk-work) [inline src/main/java/io/vertx/json/schema/impl/SchemaValidatorImpl.java:603]: I think ig the property isn't matching the value, it is more a INVALID_VALUE than a MISSING_VALUE, what do you think?

**Review by pk-work — CHANGES_REQUESTED** [review]

File: src/main/java/io/vertx/json/schema/impl/SchemaValidatorImpl.java (hunk 1 above)
Comment (CheesyBoy123) [reply src/main/java/io/vertx/json/schema/impl/SchemaValidatorImpl.java]: I think you are right here, I've updated it to be INVALID_VALUE now. Good catch thanks!

**Review by pk-work — APPROVED** [review]

**Review by pk-work — COMMENTED** [review]
Should we mention it in the Docs?

**Comment by CheesyBoy123** [comment]
Probably the best to have some documentation. I've added a little bit to it, let me know what you think!

**Pull request merged**